### PR TITLE
Add new level: Good Samaritan

### PR DESCRIPTION
**Pull request by ericnordelo**
This level shows the user that Reverts Errors can be simulated from the call stack (bubbled-up), potentially provoking unexpected behavior if the developer assumes that the Error comes from one specific contract.

In this specific example (the level), an attacker can steal all the funds "simulating" the NotEnoughBalance() error defined in the Wallet contract.

I also added support for Solidity ^0.8.0 by aliasing the library to "openzeppelin-contracts-08" using:
`yarn add openzeppelin-contracts-08@npm:@openzeppelin/contracts`

Now the game supports uploading levels in versions 0.5.3, 0.6.12, or 0.8.12.

I hope you guys find this level useful.

### CODE REVIEW

File: client/src/constants.js [hunk 1]
```diff
@@ -62,7 +62,7 @@ export const PATH_STATS = `${PATH_ROOT}stats`
 export const CUSTOM_LOGGING = true                              /* TRUE on production */
 export const SHOW_ALL_COMPLETE_DESCRIPTIONS = false             /* FALSE on production */
 export const SHOW_VERSION = true                                /* TRUE on production */
-export const ACTIVE_NETWORK = NETWORKS.RINKEBY
+export const ACTIVE_NETWORK = NETWORKS.LOCAL
```
Comment (xaler5) [inline client/src/constants.js:65]: Bring RINKEBY back

File: contracts/test/levels/GoodSamaritan.test.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    assert.equal(await coin.balances(player), 0)
+
+    // Ensure that players can requestDonation
+    await instance.requestDonation({from: player});
+
+    assert.equal((await coin.balances(wallet.address)).toNumber(), 10**6 - 10)
+    assert.equal(await coin.balances(player), 10)
+
+    // Factory check (should fail)
+    let completed = await utils.submitLevelInstance(
```
Comment (xaler5) [inline contracts/test/levels/GoodSamaritan.test.js:46]: Normally, unit tests are separated into:
- User can't solve the level directly (which is this current line)
- User can solve the level (by performing the attack).

Would you mind separating into two separate tests ?

File: contracts/test/levels/GoodSamaritan.test.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    ethernaut = await Ethernaut.new()
+    level = await GoodSamaritanFactory.new()
+    await ethernaut.registerLevel(level.address)
+  });
+
+  it('should allow the player to solve the level', async function() {
+
+    const instance = await utils.createLevelInstance(
+      ethernaut, level.address, player, GoodSamaritan,
+      {from: player, value: web3.utils.toWei('0.001', 'ether')}
```
Comment (xaler5) [inline contracts/test/levels/GoodSamaritan.test.js:27]: is 0.001 ether needed ?

File: client/src/constants.js (hunk 1 above)
Comment (ericnordelo) [reply client/src/constants.js]: Done

File: contracts/test/levels/GoodSamaritan.test.js (hunk 2 above)
Comment (ericnordelo) [reply contracts/test/levels/GoodSamaritan.test.js]: Done

File: contracts/test/levels/GoodSamaritan.test.js (hunk 3 above)
Comment (ericnordelo) [reply contracts/test/levels/GoodSamaritan.test.js]: Is not. Refactored.

**Comment by xaler5** [comment]
Good to go :) thanks @ericnordelo

**Review by xaler5 — APPROVED** [review]

**Pull request merged**